### PR TITLE
Support JEP 445 main methods

### PR DESCRIPTION
**Pull request by akurtakov**
## What it does
Using isMainMethodCandidate ensures there is at least one main method so JVM will be able to launch.

## How to test
Create a class with public/protected/default non-static main method with or without String[] args and verify that Run As/Debug As works.

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by akurtakov** [comment]
One step further in making https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1106 usabel in the IDE.

**Comment by iloveeclipse** [comment]
@akurtakov : can you please add a test that launches "simple modern" main? It should fail without your patch.

**Comment by akurtakov** [comment]
So there are no specific test plugins for debug.ui or launching plugins. Do I assume correct that https://github.com/eclipse-jdt/eclipse.jdt.debug/blob/master/org.eclipse.jdt.debug.tests is the place to put such tests or should I start new dedicated bundles?

**Comment by akurtakov** [comment]
Some hints where exactly to put such tests in the current structure would be much appreciated.

**Comment by iloveeclipse** [comment]
> Some hints where exactly to put such tests in the current structure would be much appreciated.

Somewhere at `org.eclipse.jdt.debug.tests.launching`

**Comment by akurtakov** [comment]
This proves that instance main methods are called. I don't think it's needed to repeat the extensive tests as done in jdt.core as the debug code actually calls isMainMethodCandidate which is extensively tested in core.

**Comment by iloveeclipse** [comment]
> This proves that instance main methods are called. I don't think it's needed to repeat the extensive tests as done in jdt.core as the debug code actually calls isMainMethodCandidate which is extensively tested in core.

Sure. Unfortunately both new tests fail: https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-326/6/#showFailuresLink

**Comment by akurtakov** [comment]
> Sure. Unfortunately both new tests fail: https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-326/6/#showFailuresLink

I'm trying to figure out why it's not using Java 21 .

**Comment by akurtakov** [comment]
So the failure is due to:
```
Loading trusted files from base branch master at dc3a1d9b6ba09c2931ad39487bfa4506fe360152 rather than f44e7e2fde7b33b39eb257bff065ba89859f2c01
Obtained Jenkinsfile from dc3a1d9b6ba09c2931ad39487bfa4506fe360152
‘Jenkinsfile’ has been modified in an untrusted revision
```
As I not a committer my change to Jenkinsfile is simply ignored. @jjohnstn @rgrunber Would you please push the Jenkinsfile change in its own PR so I care rebase this one on top of it?

**Comment by akurtakov** [comment]
@jjohnstn Thanks for the fast jenkinsfile change.
This PR is ready for review now.

**Comment by akurtakov** [comment]
@iloveeclipse As you already looked at this one do you want to finish the review?

**Review by iloveeclipse — APPROVED** [review]
One cosmetic issue in the pom, the rest looks fine.

File: org.eclipse.jdt.debug.tests/pom.xml [hunk 1]
```diff
@@ -74,5 +74,31 @@
 			</plugins>
 		</build>
 	</profile>
```
[the remark this answers is not in the document]
Comment (akurtakov) [reply org.eclipse.jdt.debug.tests/pom.xml]: Done.

**Pull request merged**